unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/ChatWidgetHUD/ChatHUDView.cs

```diff
@@ -242,13 +242,14 @@ public virtual void SetEntry(string messageId, ChatEntry chatEntry, bool setScro
             scrollRect.verticalNormalizedPosition = 0;
     }
 
-    public void RemoveFirstEntry()
+    public void RemoveOldestEntry()
     {
         if (entries.Count <= 0) return;
         var firstEntry = GetFirstEntry();
-        if (firstEntry == null) return;
-        Destroy(firstEntry.gameObject);
+        if (!firstEntry) return;
         entries.Remove(firstEntry.Model.messageId);
+        // must use immediate, for some reason the object is not getting destroyed with Destroy(firstEntry.gameObject)
+        DestroyImmediate(firstEntry.gameObject);
```

Comment:
We finally were going to use `Destroy` instead of `DestroyImmediate`, no?
```suggestion
        Destroy(firstEntry.gameObject);
```

Reply:
yes! messages are actually being removed because there is one frame in between